src/main/resources/assets/js/app/inputtype/ui/text/HtmlEditor.ts

```diff
@@ -1015,6 +1010,61 @@ class HtmlEditorConfigBuilder {
         return deferred.promise;
     }
 
+    private initCustomStyleSet() {
+        if (CKEDITOR.stylesSet.registered['custom']) {
+            return;
+        }
+
+        const customStyleSet: any[] = [];
+
+        customStyleSet.push({name: 'Normal', element: 'p'});
+
+        this.getAllowedHeadings().forEach((heading: string) => {
+            const headingStyleObject: any = this.headingToStyleObject(heading);
+
+            if (headingStyleObject) {
+                customStyleSet.push(headingStyleObject);
+            }
+        });
+
+        customStyleSet.push({name: 'Normal (DIV)', element: 'div'});
+        customStyleSet.push({name: 'Formatted', element: 'pre'});
+
+        if (!this.isToolDisabled('Code')) {
+            customStyleSet.push({name: 'Code', element: 'code'});
+        }
+
+        CKEDITOR.stylesSet.add('custom', <any>customStyleSet);
+    }
+
+    private headingToStyleObject(heading: string): any {
+        if (heading === 'h1') {
+            return {name: 'Heading 1', element: 'h1'};
+        }
+
+        if (heading === 'h2') {
+            return {name: 'Heading 2', element: 'h2'};
+        }
+
+        if (heading === 'h3') {
+            return {name: 'Heading 3', element: 'h3'};
+        }
+
+        if (heading === 'h4') {
+            return {name: 'Heading 4', element: 'h4'};
+        }
+
+        if (heading === 'h5') {
+            return {name: 'Heading 5', element: 'h5'};
+        }
+
+        if (heading === 'h6') {
+            return {name: 'Heading 6', element: 'h6'};
+        }
```

Comment:
Very important that we localise all the options (also the ones created in `initCustomStyleSet`), otherwise we'll lose translations:

Current (norwegian language):
![image](https://user-images.githubusercontent.com/11612490/82792314-7e8dc480-9e6f-11ea-8b0b-4731df214f40.png)

New version:
![image](https://user-images.githubusercontent.com/11612490/82792362-949b8500-9e6f-11ea-8cbd-ca39dda5f3cc.png)

**phrases.properties:**
```
  text.htmlEditor.styles.heading = Heading {0}
```
```
private headingToStyleObject(heading: string): any {
  const headingIndex = heading.match(/\d/g)[0];
  return {name: i18n('text.htmlEditor.styles.heading', headingIndex), element: heading};
}
```

Reply:
Ok!